protocol/routing/README.md

```diff
@@ -0,0 +1,93 @@
+IPFS Routing Protocol Spec
+==========================
+
+Authors: David Dias
+
+Reviewers:
+
+TODOS:
+
+-----------------------
+
+> This spec defines the routing protocol spec, covering `Peer discovery`, `Routing` and the `DHT`. The spec is a **Work In Progress**.
```

Comment:
We should separate out the DHT from Routing. Routing is an interface, that is satisfied by DHT.